core/federated/RTI/rti.c

```diff
@@ -826,8 +826,10 @@ void handle_time_advance_notice(federate_t* fed) {
     // Check downstream federates to see whether they should now be granted a TAG.
     // To handle cycles, need to create a boolean array to keep
     // track of which upstream federates have been visited.
-    bool* visited = (bool*)calloc(_RTI.number_of_federates, sizeof(bool)); // Initializes to 0.
-    send_downstream_advance_grants_if_safe(fed, visited);
+    for (int i = 0; i < fed->num_downstream; i++) {
```

Comment:
I think this change cannot possibly solve the problem we were seeing and contradicts the documentation for the `send_advance_grant_if_safe` function, which says:
```
 * This should be called whenever an immediately upstream federate sends to
 * the RTI an LTC (Logical Tag Complete), or when a transitive upstream
 * federate sends a TAN (Time Advance Notice) or a NET (Next Event Tag) message.
 * It is also called when an upstream federate resigns from the federation.
```
This change will result in the transitive part of this not being done upon receiving a TAN.

Reply:
Yes I agree. This change did not fix the issue anyway. I will revert this change. Please note that lf-lang/lingua-franca#1074 depends on this branch in reactor-c.